src/editor/Model/WindowManager.re

```diff
@@ -0,0 +1,117 @@
+open Revery_UI;
+
+module WindowSplitId =
+  Revery.UniqueId.Make({});
+
+[@deriving show({with_path: false})]
+type componentCreator = unit => React.syntheticElement;
+
+[@deriving show({with_path: false})]
+type direction =
+  | Horizontal
+  | Vertical;
+
+[@deriving show({with_path: false})]
+type split = {
+  id: int,
+  parentId: int,
+  component: componentCreator,
+  direction,
+  /* if omitted the split will grow to occupy whatever space is available */
+  width: option(int),
+  height: option(int),
+};
+
+[@deriving show({with_path: false})]
+type dock = {
+  component: componentCreator,
+  width: option(int),
+  height: option(int),
+};
+
+[@deriving show({with_path: false})]
+type splitTree =
+  | Parent(direction, int, list(splitTree))
+  | Leaf(split);
+
+[@deriving show({with_path: false})]
+type splits = splitTree;
+
+type t = {
+  windows: splits,
+  leftDock: option(dock),
```

Comment:
Note To Self: Didn't think/realise that these might have to be a list to accomodate the file explorer